examples/mytestlib.c

```diff
@@ -22,23 +22,91 @@
 #include <fcntl.h>
 #include <unistd.h>
 
-struct vaccel_arg {
-	uint32_t len;
-	void *buf;
+#include <vaccel.h>
+
+struct mydata
+{
+	uint32_t size;
+	int* array;
 };
 
+void* ser(void* buf, uint32_t* bytes){
+	struct mydata* non_ser = (struct mydata*)buf;
+
+	uint32_t size = (non_ser->size + 1) * sizeof(int);
+	int* ser_buf = malloc(size);
+
+	memcpy(ser_buf, (int*)(&non_ser->size), sizeof(int));
+
+	for(int i=0; i<(int)non_ser->size; i++)
+		memcpy(&ser_buf[i+1], &non_ser->array[i], sizeof(int));
+
+
+	*bytes = size;
+	return ser_buf;
+}
+
+
+void* deser(void* buf, uint32_t bytes){
+	int *ser_buf = (int*)buf;
+	int size = ser_buf[0];
+
+	struct mydata* new_buf =
+		(struct mydata*)malloc(sizeof(struct mydata));
+
+	new_buf->size = (uint32_t)size;
+	new_buf->array = malloc(new_buf->size * sizeof(int));
+	for(int i=0; i<size; i++)
+		memcpy(&new_buf->array[i], &ser_buf[i+1], sizeof(int));
+
+	return new_buf;
+}
+
 /* We know we're getting only one read and only one write argument */
 
 int mytestfunc(struct vaccel_arg *input, size_t nr_in,
 	       struct vaccel_arg *output, size_t nr_out)
 {
-	int a = *(int *)input[0].buf;
 	assert(nr_in >= 1);
 	assert(nr_out >= 1);
+
+	// Input 
+	int* input_int = vaccel_extract_ser_arg(input, 0);
+	
 	printf("I got nr_in: %ld, nr_out: %ld\n", nr_in, nr_out);
-	printf("I got this input: %d\n", a);
-	sprintf(output[0].buf, "I got this input: %d\n", a);
-	output[0].len = strlen(output[0].buf);
```

Comment:
since we have changed the way the library works (we return an integer, not a string) we should change the `exec.c`/`exec_with_res.c`/`exec_generic.c` example files that we use that, otherwise tests will fail.